sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientOptions.java

```diff
@@ -44,6 +45,7 @@ public IdentityClientOptions() {
         Configuration configuration = Configuration.getGlobalConfiguration();
         authorityHost = configuration.get(Configuration.PROPERTY_AZURE_AUTHORITY_HOST,
             AzureAuthorityHosts.AZURE_PUBLIC_CLOUD);
+        cp1disabled = configuration.get(Configuration.PROPERTY_AZURE_IDENTITY_DISABLE_CP1, false);
```

Comment:
I believe we had guidelines that discourage configurations through environment variables only? Has that changed now?

Reply:
This scenario was a conscious decision made across all tier-1 languages to allow to disable CP1 capability via env var only without any API changes to begin with. 
This scenario is expected to be uncommon and this env var switch is added to unblock anyone looking to disable the CP1 capability on client side, as it is enabled by default.

@schaabs